tls/s2n_renegotiate.h

```diff
@@ -15,11 +15,19 @@
 
 #pragma once
 
+#include "utils/s2n_result.h"
+
 struct s2n_connection;
 struct s2n_config;
 
+extern const s2n_blocked_status S2N_BLOCKED_ON_APPLICATION_DATA;
+
 typedef enum { S2N_RENEGOTIATE_REJECT, S2N_RENEGOTIATE_ACCEPT} s2n_renegotiate_response;
 typedef int (*s2n_renegotiate_request_cb)(struct s2n_connection *conn, void *context, s2n_renegotiate_response *response);
 int s2n_config_set_renegotiate_request_cb(struct s2n_config *config, s2n_renegotiate_request_cb cb, void *ctx);
 
 int s2n_renegotiate_wipe(struct s2n_connection *conn);
+int s2n_renegotiate(struct s2n_connection *conn, uint8_t *app_data_buf, ssize_t app_data_buf_size,
```

Comment:
🤔 Will there be another PR after this that declares s2n_renegotiate in s2n.h plus adds docs?